relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -411,37 +411,45 @@ fn extract_span_metrics(
         return Err(ExtractMetricsError::InvalidTimestamp);
     }
 
-    if let Some(_spans) = event.spans.value_mut() {
-        // Collect the shared tags for all the metrics and spans on this transaction
-        let mut shared_tags = BTreeMap::new();
+    if event.spans.value().is_none() {
+        return Ok(());
+    }
```

Comment:
This one is tricky. Since we'll need mutable references to the spans, but we need to extract metrics from the event first, we can't hold a mutable reference to the spans and a reference to the event. I'll try to clean this up next week.

Reply:
@jjbayer  is right, following  
```rust
let Some(spans) = event.spans.value() else { return Ok(())};
```
works and looks much better with early return